### PR TITLE
Add mp-drop-menu component

### PR DESCRIPTION
**Pull request by rickydoar**
Menu modal with animations and stylings

### CODE REVIEW

File: lib/components/drop-menu/index.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  close_menu() {
+    switch(this.state.visibility) {
+      case VISIBILITY_CLOSED:
+      case VISIBILITY_CLOSING:
+        break;
+      case VISIBILITY_OPENING:
+        this.update({visibility: VISIBILITY_CLOSED});
+        break;
+      case VISIBILITY_OPEN:
+        this._pendingAnimations = ['fadeModalOut'];
```
Comment (jbwyme) [inline lib/components/drop-menu/index.js]: let's rename `Modal` to `Menu` everywhere here

File: lib/components/drop-menu/index.styl [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    opacity 0.9
+
+@keyframes fadeOverlayOut
+  from
+    opacity 0.9
+  to
+    transform translateY(-10px)
+    opacity 0
+  
++prefix-classes('mp-menu-')
```
Comment (jbwyme) [inline lib/components/drop-menu/index.styl]: `mp-drop-menu`

File: lib/components/drop-menu/index.styl [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  .body
+    background-color white
+    border-color white
```
Comment (jbwyme) [inline lib/components/drop-menu/index.styl]: `border 1px solid white`

Comment (rickydoar) [inline lib/components/drop-menu/index.styl]: 6657038

File: lib/components/drop-menu/index.styl (hunk 2 above)
Comment (rickydoar) [inline lib/components/drop-menu/index.styl]: 6657038

File: lib/components/drop-menu/index.js (hunk 1 above)
Comment (rickydoar) [inline lib/components/drop-menu/index.js]: 6657038

**Pull request merged**